### PR TITLE
Adding codecov support

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

This change adds the ability to run codecoverage and send the reports to codecov

## Motivation and Context

- Helps track how much of our code is covered by unit tests
- Helps to encourage contributors to write tests.

## How To Test This

This PR needs to merged first before we can get a base value to work from.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [x] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by Dillon-Brown** [comment]
Assessed

**Comment by Dillon-Brown** [comment]
Assessed

File: composer.json [hunk 1]
```diff
@@ -35,7 +35,8 @@
     "jeroendesloovere/vcard": "v1.5",
     "browserstack/browserstack-local": "dev-master",
     "consolidation/robo": "^1.0.0",
-    "mockery/mockery": "^0.9.9"
+    "mockery/mockery": "^0.9.9",
+    "symfony/yaml": "^3.4"
```
Comment (gymad) [inline composer.json]: if it's an unused lib, can you remove please

File: tests/unit/lib/SuiteCRM/Robo/Commands/CodeCoverageCommandsTest.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
+<?php
+
+use \SuiteCRM\Robo\Plugin\Commands\CodeCoverageCommands;
+
+class CodeCoverageCommandsTest extends \Codeception\Test\Unit
```
Comment (gymad) [inline tests/unit/lib/SuiteCRM/Robo/Commands/CodeCoverageCommandsTest.php]: use state check please

**Comment by gymad** [comment]
why?

**Comment by daniel-samson** [comment]
Sorry, I did't see your comments when I merged ... I have addressed them here #5862.

File: lib/Robo/Plugin/Commands/CodeCoverageCommands.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+ * reasonably feasible for technical reasons, the Appropriate Legal Notices must
+ * display the words "Powered by SugarCRM" and "Supercharged by SuiteCRM".
+ */
+namespace SuiteCRM\Robo\Plugin\Commands;
+
+use SuiteCRM\Utility\OperatingSystem;
+use SuiteCRM\Robo\Traits\RoboTrait;
+use Robo\Task\Base\loadTasks;
+use SuiteCRM\Utility\Paths;
+use Symfony\Component\Yaml\Yaml;
```
Comment (gymad) [inline lib/Robo/Plugin/Commands/CodeCoverageCommands.php]: is it used anywhere?

File: composer.json (hunk 1 above)
Comment (daniel-samson) [reply composer.json]: removed

File: lib/Robo/Plugin/Commands/CodeCoverageCommands.php (hunk 3 above)
Comment (daniel-samson) [reply lib/Robo/Plugin/Commands/CodeCoverageCommands.php]: removed

File: tests/unit/lib/SuiteCRM/Robo/Commands/CodeCoverageCommandsTest.php (hunk 2 above)
Comment (daniel-samson) [reply tests/unit/lib/SuiteCRM/Robo/Commands/CodeCoverageCommandsTest.php]: I have switched the unit to the \SuiteCRM\StateCheckerUnitAbstract

File: lib/Robo/Plugin/Commands/CodeCoverageCommands.php [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+     * Disables the state checker
+     */
+    private function disableStateChecker()
+    {
+        global $sugar_config;
+        require_once 'include/utils/file_utils.php';
+
+        $sugar_config['state_checker']['test_state_check_mode'] = 0;
+
+        return write_array_to_file(
```
Comment (gymad) [inline lib/Robo/Plugin/Commands/CodeCoverageCommands.php]: This functionality already in modules/Configurator/Configurator.php
Configurator::readOverride(), Configurator::saveOverride(), Configurator::replaceOverride() etc...

**Pull request merged**